gapic/src/main/com/google/gapid/models/Memory.java

```diff
@@ -204,7 +204,7 @@ public long getEndAddress() {
       List<ListenableFuture<Segment>> futures = Lists.newArrayList();
       futures.add(getPage(firstPage, getOffsetInPage(offset), PAGE_SIZE - getOffsetInPage(offset)));
       for (long page = firstPage + 1, left = length - PAGE_SIZE + getOffsetInPage(offset);
-          page <= lastPage; page++, left -= PAGE_SIZE) {
+          page <= lastPage && left >= 0; page++, left -= PAGE_SIZE) {
```

Comment:
The logic to compute `left` seems correct. So it seems more like a garbage-in-garbage-out issue. Instead of guarding like that, we should figure what's causing it. I think `left` is only negative if `length` is negative or 0, if that's the case, we should check that `length` is valid instead/fix callers.

Reply:
Indeed, just to confirm, when this happens, the previous call in the stack will have the `length` being 0.